core/src/main/java/org/apache/iceberg/RowLevelOperationMode.java

```diff
@@ -38,17 +38,28 @@
  * during reads.
  */
 public enum RowLevelOperationMode {
-  COPY_ON_WRITE,
-  MERGE_ON_READ;
+  COPY_ON_WRITE("copy-on-write"),
+  MERGE_ON_READ("merge-on-read");
+
+  private final String modeName;
+
+  RowLevelOperationMode(String modeName) {
+    this.modeName = modeName;
+  }
 
   public static RowLevelOperationMode fromName(String modeName) {
     Preconditions.checkArgument(modeName != null, "Mode name is null");
-    if ("copy-on-write".equalsIgnoreCase(modeName)) {
+    if (COPY_ON_WRITE.toString().equalsIgnoreCase(modeName)) {
       return COPY_ON_WRITE;
-    } else if ("merge-on-read".equalsIgnoreCase(modeName)) {
+    } else if (MERGE_ON_READ.toString().equalsIgnoreCase(modeName)) {
       return MERGE_ON_READ;
     } else {
       throw new IllegalArgumentException("Unknown row-level operation mode: " + modeName);
     }
   }
+
+  @Override
+  public String toString() {
+    return this.modeName;
```

Comment:
nit: Just `return modeName` as we usually include `this` only in setters?

Reply:
I agree with @aokolnychyi here, I think it's better to have `modeName()` than overriding `toString()`